### ISSUE TITLE
Investigation: Why does NVDA read out 'clickable' when reading images

### ISSUE THREAD

**Issue by greenc05** [bbc/simorgh#9746]
**Describe the bug**
When using NVDA with Firefox ESR on Windows 10 it reads out 'clickable' when I read images. Why is this?

**To Reproduce**
Given I am using NVDA on Windows 10 with Firefox ERS
When I press the down arrow key to read an image
Then 'clickable' is announced along with the alt text and image semantics

**Expected behaviour**
Clickable should not be announced when I read an image.

**Comment by DarioR01** [comment]
****Investigation report****
When we started investigating this bug we immediately noticed that if we open the accessibility tree in Firefox we can notice keyboard's warnings attached to all graphic elements that are announced as clickable. We also discovered that every image that was being announced as clickable had a Click action visible from the properties dropdown:

<img width="1792" alt="Screenshot 2022-01-20 at 16 03 45" src="https://user-images.githubusercontent.com/90621252/150376163-4b369273-205b-4ec6-9d6d-95c9e0d095c1.png">

From [this](https://stackoverflow.com/questions/68365778/nvda-reads-logo-as-clickable-when-it-isnt) stackOverflow's article we understood that this event was being passed down from the parent elements. But this did not explain why only graphics were announcing clickable.

We decided to go back to past PR's and understand when we introduced the bug. We noticed how the Glass Promo component was the only coponent with a graphic element that was not announcing "_clickable graphic_" in past PR's.
This motivated us for searching the PR that introduced the bug in the FrostedGlassPromo component.
We identified the following PR being the one that introduced the bug. After further investigation we discovered that if we replace [this line](https://github.com/bbc/simorgh/blob/60b88318de00e85412c877ea683d9677ab2e771f/src/app/components/FrostedGlassPromo/index.jsx#L22) with ``const Wrapper = styled.a` `` ,and [this line](https://github.com/bbc/simorgh/blob/60b88318de00e85412c877ea683d9677ab2e771f/src/app/components/FrostedGlassPromo/index.jsx#L91) with `` <Wrapper href="url" data-testid={ `frosted-promo-${index}`> `` NVDA will make it stop to announce "_clickable graphic_" .
My assumption is that the click event gets overwritten by the basic clickevent of the anchor tag. That would also explain why this fix only works when we have an href parameter passed to it.
However this has no use for us, because it makes the clickable image a link, which is not the behaviour we want for all images.

Finally we discovered the following [issue](https://github.com/facebook/react/issues/20895) reported in 2021 to facebook React.
After some testing we managed to reproduced the described bug in simorgh by modifying the [clint.js](https://github.com/bbc/simorgh/blob/latest/src/client.js) and [components.js](https://github.com/bbc/simorgh/blob/latest/src/server/Document/component.jsx) as showed from the following images:


****Conclusion****

**Comment by DarioR01** [comment]
****Investigation report****
When we started investigating this bug we immediately noticed that if we open the accessibility tree in Firefox we can notice the keyboard's warnings attached to all graphic elements that are announced as clickable. We also discovered that every image that was being announced as clickable had a Click action visible from the properties dropdown:

<img width="1792" alt="Screenshot 2022-01-20 at 16 03 45" src="https://user-images.githubusercontent.com/90621252/150376163-4b369273-205b-4ec6-9d6d-95c9e0d095c1.png">

From [this](https://stackoverflow.com/questions/68365778/nvda-reads-logo-as-clickable-when-it-isnt) StackOverflow's article we understood that this event was being passed down from the parent elements. But this did not explain why only graphics were announcing clickable.

We decided to go back to past PR's and understand when we introduced the bug. We noticed how the Glass Promo component was the only component with a graphic element that was not announcing "_clickable graphic_" in past PR's.
This motivated us for searching the PR that introduced the bug in the FrostedGlassPromo component.
We identified the following PR being the one that introduced the bug. After further investigation we discovered that if we replace [this line](https://github.com/bbc/simorgh/blob/60b88318de00e85412c877ea683d9677ab2e771f/src/app/components/FrostedGlassPromo/index.jsx#L22) with ``const Wrapper = styled.a` `` ,and [this line](https://github.com/bbc/simorgh/blob/60b88318de00e85412c877ea683d9677ab2e771f/src/app/components/FrostedGlassPromo/index.jsx#L91) with `` <Wrapper href="url" data-testid={ `frosted-promo-${index}`> `` NVDA will make it stop to announce "_clickable graphic_" .
My assumption is that the click event gets overwritten by the basic click event of the anchor tag. That would also explain why this fix only works when we have an href parameter passed to it.
However, this has no use for us, because it makes the clickable image a link, which is not the behaviour we want for all images.

Finally, we discovered the following [issue](https://github.com/facebook/react/issues/20895) reported in 2021 to facebook React.
After some testing I managed to reproduce the described bug in Simorgh by modifying the [clint.js](https://github.com/bbc/simorgh/blob/latest/src/client.js) and [components.js](https://github.com/bbc/simorgh/blob/latest/src/server/Document/component.jsx) as shown from the following images:

client.js:
![image (1)](https://user-images.githubusercontent.com/90621252/150383145-cc1f40e1-c6a9-4ce9-93dc-29e34261b6c8.png)

components.js
![image (2)](https://user-images.githubusercontent.com/90621252/150383178-1065f4ac-579d-4d7e-88a1-081799fdde96.png)

I have essentially added 2 tests image, one outside the "root" division and one just inside the root div but before any other element.
What we observed is that the one outside presented no bug, but the one inside was still announcing "clickable graphics": 

![image (1)](https://user-images.githubusercontent.com/90621252/150784564-57d28e9a-54d5-4081-bd4b-db335ca704df.png)

And we also observed that we have no keyboard warning appearing in the graphic element outside the root div:

![image](https://user-images.githubusercontent.com/90621252/150384426-60c98301-112a-46fd-8e2a-32c7085be8c6.png)

We also made a minor discovery where the keyboard warning attached to the graphic element seems to be affected by the LTR and RTL logic 😂 
perhaps in Arabic service, we would observe:

<img width="1428" alt="Screenshot 2022-01-19 at 11 57 45" src="https://user-images.githubusercontent.com/90621252/150385006-2fe0ef4b-3a51-4eae-a4fc-b8d56a47770c.png">


****Conclusion****

React 17 had changes to event delegation https://reactjs.org/blog/2020/08/10/react-v17-rc.html#changes-to-event-delegation
In React 16, the events were attached to the document instead of the root
v16 - `` document.addEventListener('click', () => { console.log('click') }) ``
v17 - `` document.querySelector('#root').addEventListener('click', () => { console.log('click') }) ``

Take this simple HTML as an example:

```
<!DOCTYPE html>
<html lang='en'>
	<head>
		<title>Basic HTML5 document</title>
		<meta charset='utf-8'>
	</head>
	<body>
		<div id="root">
			<img src="https://picsum.photos/200/300" alt="test text"/>
		</div>

		<script>
			// event delegation
			document.querySelector('#root').addEventListener('click', () => { console.log('click') })
		</script>
	</body>
</html>
```
If we run the above and inspect with NVDA we can observe the bug.
But if we attach the listener to the document instead of the root, the bug goes away.

We reported this behaviour with [Facebook React](https://github.com/facebook/react/issues/23170), [NVDA](https://github.com/nvaccess/nvda/issues/13262), and [W3C/aria](https://github.com/nvaccess/nvda/issues/13262).

**Comment by DarioR01** [comment]
Updates:

- Facebook/React closed out the ticket as it was a duplicate.
- We closed the W3C/aria ticket because we discovered new information about [clickable elements](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics) we believe the bug should be fixed by NVDA or  Firefox.
- NVDA suggested a solution where role="presentation" solves the bug we are experiencing. After some tests:

```
<div role="presentation">
        <img src="https://picsum.photos/200/300" alt="div with presentation" />
</div>

<figure role="presentation">
        <img
          src="https://picsum.photos/200/300"
          alt="figure with presentation"
        />
</figure>

<img
role="presentation"
src="https://picsum.photos/200/300"
alt="No wrap and presentation"
/>
```
We discovered that all of the above fail to fix our issue for various reasons.

However, putting role="presentation" directly in the root div seems to fix the graphic clickable bug.
We will take this into consideration as a last resource.

**Comment by DarioR01** [comment]
React are going to recommend adding role="presentation" natively in the root div https://github.com/facebook/react/issues/20895#issuecomment-1021175889

**Comment by DarioR01** [comment]
We can make the changes ourselves by https://github.com/bbc/simorgh/pull/9828

**Comment by karinathomasbbc** [comment]
cc @greenc05?

**Comment by eagerterrier** [comment]
closing as part of a clear up. Please raise a new issue if the issue persists. Thanks
